src/pybind/mgr/dashboard/frontend/src/app/shared/api/rgw-bucket.service.ts

```diff
@@ -15,11 +15,47 @@ import { cdEncode } from '~/app/shared/decorators/cd-encode';
 })
 export class RgwBucketService extends ApiClient {
   private url = 'api/rgw/bucket';
+  private bucketsSubject = new BehaviorSubject<object[]>([]);
+  private totalNumObjectsSubject = new BehaviorSubject<number>(0);
+  private totalUsedCapacitySubject = new BehaviorSubject<number>(0);
+  private averageObjectSizeSubject = new BehaviorSubject<number>(0);
+  buckets$ = this.bucketsSubject.asObservable();
+  totalNumObjects$ = this.totalNumObjectsSubject.asObservable();
+  totalUsedCapacity$ = this.totalUsedCapacitySubject.asObservable();
+  averageObjectSize$ = this.averageObjectSizeSubject.asObservable();
 
   constructor(private http: HttpClient, private rgwDaemonService: RgwDaemonService) {
     super();
   }
 
+  fetchAndTransformBuckets() {
+    return this.list(true).pipe(
+      map((buckets: object[]) => {
+        let totalNumObjects = 0;
+        let totalUsedCapacity = 0;
+        let averageObjectSize = 0;
+        const transformedBuckets = buckets.map((bucket) => {
+          const maxBucketSize = bucket['bucket_quota']['max_size'];
+          const maxBucketObjects = bucket['bucket_quota']['max_objects'];
+          bucket['bucket_size'] = bucket['usage']?.['rgw.main']?.['size_actual'] || 0;
+          bucket['num_objects'] = bucket['usage']?.['rgw.main']?.['num_objects'] || 0;
+          bucket['size_usage'] =
+            maxBucketSize > 0 ? bucket['bucket_size'] / maxBucketSize : undefined;
+          bucket['object_usage'] =
+            maxBucketObjects > 0 ? bucket['num_objects'] / maxBucketObjects : undefined;
+          totalNumObjects += bucket['num_objects'];
+          totalUsedCapacity += bucket['bucket_size'];
+          averageObjectSize = totalNumObjects > 0 ? totalUsedCapacity / totalNumObjects : 0;
+          return bucket;
+        });
```

Comment:
We may want to break this into smaller functions for better readability

Reply:
Done, thanks